blocks/init/src/Blocks/custom/featured-categories-posts/featured-categories-posts.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+/**
+ * Template for the Featured Categories Posts view.
+ *
+ * @package EightshiftBoilerplate
+ */
+
+use EightshiftBoilerplateVendor\EightshiftLibs\Helpers\Components;
+
+$manifest = Components::getManifest(__DIR__);
+
+$blockClass = Components::checkAttr('blockClass', $attributes, $manifest);
+$query = Components::checkAttr('query', $attributes, $manifest);
+$excludeCurrentPost = Components::checkAttr('excludeCurrentPost', $attributes, $manifest);
+$itemsPerLine = Components::checkAttr('itemsPerLine', $attributes, $manifest);
+$showItems = Components::checkAttr('showItems', $attributes, $manifest);
+$serverSideRender = Components::checkAttr('serverSideRender', $attributes, $manifest);
+
+global $post;
+
+?>
+
+<div class="<?php echo esc_attr($blockClass); ?>" data-items-per-line=<?php echo \esc_attr($itemsPerLine); ?>>
+
+		<?php
+		$postType = $query['postType'];
+		$taxonomy = $query['taxonomy'];
+		$terms = $query['terms'];
+
+		$args = [
+			'post_type' => $postType,
+			'posts_per_page' => $showItems,
+		];
+
+		if ($taxonomy) {
+			if ($terms) {
+				$args['tax_query'][0] = [
+					'taxonomy' => $taxonomy,
+					'field' => 'id',
+					'terms' => $terms,
+				];
+			} else {
+				$args['tax_query'] = [ // phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_tax_query
+					[
+						'taxonomy' => $taxonomy,
+					],
+				];
+			}
+		};
+
+		if ($excludeCurrentPost) {
+			$args['post__not_in'] = [ $post->ID ];
+		}
+
+		$theQuery = new \WP_Query($args);
```

Comment:
I'm not thrilled by the taxonomy queries or the `post__not_in`, as these are generally slow queries.

But we can profile later on to see if this is a bottleneck and come up with a solution later on.